fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/connection/FHIRDbProxyDatasourceConnectionStrategy.java

```diff
@@ -0,0 +1,168 @@
+            }
+        }
+    }
+
+    /* (non-Javadoc)
+     * @see com.ibm.fhir.persistence.jdbc.connection.FHIRDbConnectionProvider#getConnection()
+     */
+    @Override
+    public Connection getConnection() throws FHIRPersistenceDBConnectException {
+        Connection connection;
+        final String METHODNAME = "getConnection";
+
+        if (log.isLoggable(Level.FINE)) {
+            log.entering(CLASSNAME, METHODNAME);
+        }
+
+        try {
+            // Resources can be routed to different databases using the dsId currently
+            // set on the context.
+            String tenantId = FHIRRequestContext.get().getTenantId();
+            String dsId = FHIRRequestContext.get().getDataStoreId();
+            
+            long start = System.nanoTime();
+            if (log.isLoggable(Level.FINE)) {
+                log.fine("Getting connection for tenantId/dsId: [" + tenantId + "/" + dsId + "]...");
+            }
+
+            // Use the username/password interface to pass the tenantId and datasource id
+            // paramters into the proxy datasource so that it can find the correct connection
+            connection = getConnection(datasource, tenantId, dsId);
+
+            if (log.isLoggable(Level.FINE)) {
+                double deltams = (System.nanoTime() - start) / NANOMS;
+                log.fine("Got the connection for [" + tenantId + "/" + dsId + "]. Took " + deltams + " ms");
+            }
+        } catch (Throwable e) {
+            // Don't emit secrets in case they are returned to a client
+            FHIRPersistenceDBConnectException fx =
+                    new FHIRPersistenceDBConnectException("Failure acquiring connection for datasource");
+            throw FHIRDbHelper.severe(log, fx, "Failure acquiring connection for datasource: " + datasourceJndiName, e);
+        } finally {
+            if (log.isLoggable(Level.FINE)) {
+                log.exiting(CLASSNAME, METHODNAME);
+            }
+        }
+        
+        return connection;
+    }
+    
+    @Override
+    protected Connection getConnection(DataSource datasource, String tenantId, String dsId) throws SQLException, FHIRPersistenceException {
+        // Now use the dsId/tenantId specific JEE datasource to get a connection
+        Connection connection = datasource.getConnection(tenantId, dsId);
+        
+        try {
+            // always
+            connection.setAutoCommit(false);
+            
+            // configure the connection if it's the first time we've accessed it in this transaction
+            configure(connection, tenantId, dsId);
+        } catch (Throwable t) {
+            // clean up if something goes wrong during configuration
+            try {
+                connection.close();
+            } catch (Throwable x) {
+                // NOP...something bad is going on anyway, so don't confuse things
+                // by throwing a different exception and hiding the original
```

Comment:
I still think it's appropriate to log finest

Reply:
which bit? Perhaps we should create some guidelines for FINE/FINER/FINEST.

Reply:
if you are catching a throwable here, it'd be a great think to have a log statement at least announcing.  I say FINEST as it is not FINER which is our default, and allows us to turn it higher to see what's happening.

Reply:
fixed in next commit